tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -8483,4 +8483,77 @@ public function testFormValuesSourcesResetViaEnd()
         $result = $this->Form->getValuesSources();
         $this->assertEquals($expected, $result);
     }
+
+    /**
+     * Test sources values schema defaults handling
+     *
+     * @return void
+     */
+    public function testFormValuesSourcesSchemaDefaults()
+    {
+        $Articles = TableRegistry::get('Articles');
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $result = $this->Form->getSourceValue('title');
+        $expected = '';
+        $this->assertEquals($expected, $result);
+
+        $Articles = TableRegistry::get('Articles');
+        $title = $Articles->schema()->column('title');
+        $Articles->schema()->addColumn(
+            'title',
+            ['default' => 'default title'] + $title
+        );
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $result = $this->Form->getSourceValue('title');
+        $expected = 'default title';
+        $this->assertEquals($expected, $result);
+
+        $Articles = TableRegistry::get('Articles');
+        $title = $Articles->schema()->column('title');
+        $Articles->schema()->addColumn(
+            'title',
+            ['default' => 'default title'] + $title
+        );
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $this->Form->setValuesSources(['query']);
+        $result = $this->Form->getSourceValue('title');
+        $expected = '';
+        $this->assertEquals($expected, $result);
```

Comment:
> Why don't the schema defaults apply when the source is 'query'?

@markstory  Because only context instance knows about the schema defaults. So just setting sources to `['query']` wouldn't would apply schema defaults but setting it to `['query', 'context']` would.